### ISSUE TITLE
Allow the duration parameter to be a np.int in contrib.handler

### ISSUE THREAD

**Issue by zippeurfou** [pytorch/ignite#442]
I had some issue with concat scheduler using numpy integer for durations instead of int:
<img width="1145" alt="image 2019-02-20 at 1 54 36 pm" src="https://user-images.githubusercontent.com/4520383/53116736-58180580-3517-11e9-9634-6dcacd0c90c4.png">
this seems pretty restrictive when you could just do something along theses lines:
```
all([isinstance(t, (int,np.integer)) for t in durations])
```
Happy to submit a PR if you think it makes sense.

**Comment by vfdev-5** [comment]
@zippeurfou can you post the code that fails, please.
Maybe, it would be better to compare with `numbers.Integral`:
```python
import numbers

a = np.array([0, 1, 2, 3])
print(isinstance(a[0], int), isinstance(a[0], numbers.Integral))
> False, True
```

**Comment by zippeurfou** [comment]
Sure:
```
import numpy as np
import torch
trainer = Engine(lambda engine, batch: None)
optimizer = torch.optim.Adam([tensor], betas=(0.9, 0.99))
scheduler = LinearCyclicalScheduler(optimizer, 'lr', 0.01,0.3,cycle_size=10) 
#this works
sc = ConcatScheduler(schedulers=[scheduler,scheduler],durations=[30,])
#this fails
sc = ConcatScheduler(schedulers=[scheduler,scheduler],durations=[np.floor(30).astype(np.int),])
```

**Comment by vfdev-5** [comment]
Above example looks rather artificial :)
Is it practical to set `durations` as a numpy array?

**Comment by zippeurfou** [comment]
The actual practical example is when I was calculating the duration by doing something like this:
`np.floor(steps_per_epoch/2)`.I could have used the math library for it but there are case where you want to define your cycle_size, duration programmatically.
This is definitely a minor issues.

**Comment by vfdev-5** [comment]
@zippeurfou I see, make sense. Even if we can probably set it like `steps_per_epoch // 2` (integral division).
Anyway, we can replace all `isinstance(a, int)` by `isinstance(a, numbers.Integral)` to be more safe.
So, if you would like to send a PR, you're welcome :)
